### PR TITLE
Added worldfengur scraper

### PR DESCRIPTION
**Pull request by Loknar**
Scraping the following page: http://www.worldfengur.com/freezone_horse.jsp?c=EN
Search parameters that can be provided: id, name, origin, microchip.
One must provide at least one of the following: id, name and origin, or microchip.
Most of the time we receive results with list of a one single horse (or empty list if no horse was found).
There are edge cases where more than one horse can be found for a given search, when multiple horses share a combination of name and origin.

Implementation should be async as far as my understanding of it goes, I have some but perhaps limited experience with async javascript.
There is at least one hack here (to encode utf8 string to ISO 8859-1, without the "%25"), the page expects ISO 8859-1 coded text, please unhack this if you know a better way to do this.

### CODE REVIEW

**Comment by koddsson** [comment]
@benediktvaldez @Loknar talked to me IRL and was wondering where the docs live now? I keep forgetting D:

**Comment by koddsson** [comment]
@Loknar I'm gonna jump in to this and change some stuff around if that's cool with you. I promise I'll put a decent description into the commit and you can ask about it here if something I'm doing is unclear.

**Comment by Loknar** [comment]
@koddsson Ok thanks :)

**Comment by Loknar** [comment]
@koddsson in https://github.com/apis-is/apis/pull/273/commits/8c55e0c13850311ae16d44d5129dc7e45a5748fa the function `asyncLoop` was removed and a map function used in its place, this makes multi-records scenarios as I had implemented them fail.

The `asyncLoop` was used to do single-result query for each horse we got in multi-results scenario.
The reason for doing this is that we get more data in single-result scenario than multi-results scenario.

I've re-added the `asyncLoop` implementation, I like it because I believe the multi-results cases are both very rare and when they do come up we get 2 or 3 horses max, so it's not really expensive to do extra queries for them to get as complete data as we can? Of course I'm willing to admit it's perhaps wiser to not do extra queries and just return the data we got in the first query for multi-results. Thoughts?

Also thanks again for reviewing. :+1:

**Comment by koddsson** [comment]
@Loknar I think you are correct in changing it back. The homemade `asyncLoop` seemed off to me when refactoring but I also wasn't 100% on all the code since I didn't write it :smile:

File: endpoints/horses/index.js [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+    {name: 'date_of_birth',   label: 'Date of birth',               value: null},
+    {name: 'color_code',      label: 'Colour code',                 value: null},
+    {name: 'color',           label: 'Colour ',                     value: null},
+    {name: 'country_located', label: 'Country of current location', value: null},
+    {name: 'fate',            label: 'Fate ',                       value: null},
+    {name: 'microchip',       label: 'Microchip',                   value: null},
+    {name: 'father',          label: 'Sire',                        value: null},
+    {name: 'mother',          label: 'Dam',                         value: null},
+  ];
+  var labels = params.map(function(x){return x['label'];});
```
Comment (koddsson) [inline endpoints/horses/index.js]: :clap: :clap: :clap:

Comment (koddsson) [inline endpoints/horses/index.js]: In recent versions of node we have [arrow functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions) so we can now write anonymous functions like this:

```js
var labels = params.map(x => x.label);
```

which is a little bit cleaner, but don't bother changing this now unless you really want. I'm working on getting our code linted and using this syntax in the `eslint` branch. (You can check it's progress in #274 if you are interested).

**Comment by koddsson** [comment]
This looks good to me!

**Comment by koddsson** [comment]
Thank you for this @Loknar! :pray:

**Comment by Loknar** [comment]
:muscle:

**Pull request merged**